src/api/auth/LighthouseAuthApi.ts

```diff
@@ -35,15 +59,36 @@ export class LighthouseAuthApi implements AuthApi {
     username?: string,
     password?: string
   ): Promise<User | null> {
-    throw new Error('Method not implemented.');
+    const apiSignUpResponse = await fetch(`${this.url}/register`, {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        username,
+        password,
+        registration_key: registrationKey,
+        email: "todo@example.com" // TODO: add email to parameters
+      }),
+    });
+    const apiUser: ApiUser = await apiSignUpResponse.json();
+    const user: User = {
+      username: apiUser.username,
+      role: undefined, // TODO: change role to roles
```

Comment:
Can be omitted, values for any unspecified keys are implicitly `undefined`

Reply:
I left that in explicitly as a reminder that we have to handle the users roles later

Reply:
But you can remove it, if you want :smiley:

Reply:
Makes sense, let's keep this then